mlflow/utils/databricks_utils.py

```diff
@@ -84,22 +84,46 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         self.db_utils.notebook.entry_point.clearMlflowProperties()
 
 
-def _get_dbutils():
+class _NoDbutilsError(Exception):
+    pass
+
+
+class _NoIPythonShell(Exception):
+    pass
+
+
+def _get_ipython_shell():
     try:
         import IPython
 
         ip_shell = IPython.get_ipython()
         if ip_shell is None:
-            raise _NoDbutilsError
-        return ip_shell.ns_table["user_global"]["dbutils"]
+            raise _NoIPythonShell()
+        return ip_shell
     except ImportError:
-        raise _NoDbutilsError
+        raise _NoIPythonShell()
+
+
+def _get_dbutils():
+    try:
+        ip_shell = _get_ipython_shell()
+        return ip_shell.ns_table["user_global"]["dbutils"]
     except KeyError:
-        raise _NoDbutilsError
+        raise _NoDbutilsError()
 
 
-class _NoDbutilsError(Exception):
-    pass
+def _get_databricks_repl_context():
+    dbr_major_minor_version = get_databricks_runtime_major_minor_version()
+    if dbr_major_minor_version >= (14, 1):
```

Comment:
i think with the change in https://github.com/mlflow/mlflow/pull/11306, this comparison might fail if image key is "client" (comparing int and string)

Reply:
Good catch. I am going to fix this in Databricks Runtime instead. So that we don't need patch release

Reply:
Filed https://github.com/databricks/universe/pull/510153 instead.